zyte_common_items/pipelines.py

```diff
@@ -53,3 +57,43 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropItem(Exception):
+    """Drop item from the item pipeline."""
+
+    pass
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, stats):
+        self.stats = stats
+        self.threshold = None
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        o = cls(crawler.stats)
+        return o
+
+    def get_threshold(self, item, spider):
+        return spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {}).get(
+            global_object_name(item.__class__), DEFAULT_ITEM_PROBABILITY_THRESHOLD
+        )
```

Comment:
We could read this setting in `__init__` and pre-process all keys, so that if they are not `str` we use `global_object_name` on them. That way, in code, using the classes as keys instead of their import path will also work.